app/models/app_resource_monitor/hud_references_inspector.rb

```diff
@@ -0,0 +1,119 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# report on invalid HUD references
+class AppResourceMonitor::HudReferencesInspector
+  def self.enrollment_references
+    new.enrollment_references
+  end
+
+  def self.project_references
+    new.project_references
+  end
+
+  def self.client_references
+    new.client_references
+  end
+
+  def self.duplicate_ids
+    new.duplicate_ids
+  end
+
+  def hud_models
```

Comment:
Should the results of this be stored in an instance variable? It seems unnecessary to recalculate it (and re-eager-load) for each reference check call.

Reply:
Yes, memoizing this would make the it marginally more performant; probably save a few 100ms? I would totally do that if this was used in a request context. But this is called from a cron at long intervals so it's hard for me to justify the time for micro optimizations